### ISSUE TITLE
Can deduce object appearance based on the glyph

### ISSUE THREAD

**Issue by Tomsod** [NetHack/NetHack#305]
When an object which appearance is unknown is displayed on the map, it will only be described in general ("a potion", "a shield", etc.), but many such objects will still have an unique glyph, from which their appearance can be guessed (by glyph color on tty, or by the picture when using tiles).  This is mostly relevant in three cases:
* When attempting the zen conduct.
* When casting _detect treasure_ at Unskilled or Basic (or drinking the uncursed potion).
* When casting _detect treasure_ at any level, and the item is in a monster's inventory.

This is clearly an exploit that sidesteps appearance being unknown in many cases: e.g., in a zen game, if the wand of cancellation is wooden, one can safely put all non-brown wands in a bag of holding; or, if the player detects a red "! (a potion)" near a monster, and potions of gain level are ruby, the monster must have one and should be killed quickly.

This can be solved by adding grayscale glyphs for a "generic wand", "generic potion", etc.  The objects affected by this exploit are potions, spellbooks, gems, wands (especially with tiles), rings (especially with tiles), some shields, and with the default tileset, scrolls of blank paper, scrolls of mail, and rocks.

**Comment by nhmall** [comment]
yes. I think it dates back to Hack, no?

add gray stones with tiles to the list too, although stock tile sets avoid exploitation of that and keep them identical, as they should

**Comment by Tomsod** [comment]
Actually, both gray stones and ordinary rocks (which have a different tile) are described as "stones" when blind, so it does apply to rocks at least.

EDIT: oh, and also all gems!  I totally forgot gems.

**Comment by FredrIQ** [comment]
Hack did not colorize tiles, so it doesn't suffer from this problem. But yes, it's a very old problem nevertheless.

**Comment by nhmall** [comment]
Possibly related:
85c908cb038bc9b8793568f12b15e03c80828cdd
2e1f52e88236d8d0c241e4f0d397512234d92804 
68d8b5376a6638bc24cbc4efc5a5c2ca9567d4ae 
042d40faac6602128da94d02aebbb8f75b49c3b8

**Comment by Tomsod** [comment]
Yes, 85c908c definitely goes towards fixing this issue, but I think wands and rings should be affected too.  At least wands are important because of zen games and cancellation, as I wrote above.
